lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -500,8 +500,6 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
                            SILType type, DeclContext *DeclCtx,
                            GenericEnvironment *GE) {
     auto RealType = type.getASTType();
-    if (type.isAddress())
-      RealType = CanInOutType::get(RealType);
```

Comment:
This worries me. Are you representing the address-ness some other way now?

Reply:
I think this code was unused.

Reply:
But I didn't write this, so I guess @adrian-prantl is in a better position to judge.

Reply:
We need to check whether LLDB attaches any semantics to InOut or if it is pure sugar. If it's the latter then this is safe.

Reply:
Do we have any tests in LLDB that exercise inout types?

Reply:
This part worries me because it *isn't* about `inout`; it's about SIL address-only types.

Reply:
@adrian-prantl We do have some coverage:
```
Davides-Mac-Pro:testcases davide$ grep -R 'inout' * | wc -l
      29
```

@jrose-apple I'm not particularly familiar with this, can you provide an example?

Reply:
I'll try this one and I'll let you know. I do think it's an useful addition for the lldb test suite anyway, as we don't have coverage for that, so I'll add it.

Reply:
I'd put a regular Int field in the struct too, then, and make sure you can print that. That's easier than checking the weak reference, though of course that should also be correct.

Reply:
@jrose-apple this is what I'm seeing.

```
(lldb) target create "./run"
b -l Current executable set to './run' (x86_64).
(lldb) b -l 8
Breakpoint 1: where = run`run.test(run.X) -> () + 50 at run.swift:8:9, address = 0x0000000100001742
(lldb) r
Process 16896 launched: '/Users/davide/build/Ninja-ReleaseAssert+stdlib-Release/swift-macosx-x86_64/bin/run' (x86_64)
Process 16896 stopped
* thread #1, queue = 'com.apple.main-thread', stop reason = breakpoint 1.1
    frame #0: 0x0000000100001742 run`test(x=run.X @ 0x00007ffeefbff9b0) at run.swift:8:9
   5   	
   6   	func test(_ x: X) {
   7   	  var y = x
-> 8   	  print(y)
   9   	}
   10  	
   11  	test(X(foo: 23 as AnyObject, patatino: 23))
Target 0: (run) stopped.
(lldb) frame var y
(run.X) y = {
  foo = nil
  patatino = 23
}
```

```
struct X {
  weak var foo: AnyObject?
  var patatino : Int
}

func test(_ x: X) {
  var y = x
  print(y)
}

test(X(foo: 23 as AnyObject, patatino: 23))
```

Reply:
That looks like correct behavior, although it would be a better test if you had a class instance that was kept alive until after you were done calling `test`. If that works with and without your change, then I think it's all good.

Reply:
@jrose-apple Looks like that case is broken with and without my change

```
(lldb) frame var y
(run.X) y = {
  foo = (instance = 0x0000000100a00002) {
    instance = 0x0000000100a00002
  }
  patatino = 23
}
```

```
class Tinky {
  var x : Int
  init(_ x : Int) {
    self.x = x
  }
}

struct X {
  weak var foo: AnyObject?
  var patatino : Int
}

func test(_ x: X) {
  var y = x
  print(y)
}

var a = Tinky(23)
test(X(foo: a, patatino: 23))
```

Reply:
@jrose-apple given it seems unrelated, I'll open a `rdar` for this and look at it next.